packages/contracts/contracts/identity/NonFungibleRegistryUpgradeable.sol

```diff
@@ -337,6 +337,7 @@ contract NonFungibleRegistryUpgradeable is
         require(_isApprovedOrOwner(_msgSender(), tokenId), "NonFungibleRegistry: caller is not owner nor approved nor registrar.");
         require(!_mappingExists(label), "NonFungibleRegistry: label is already registered.");
 
+        delete registryMap[reverseRegistryMap[tokenId]];
```

Comment:
It will be better to check for existence of reverseRegistryMap[tokenId] in registryMap as initial token might have been created without label.